spring-test/src/main/java/org/springframework/test/web/client/match/MockRestRequestMatchers.java

```diff
@@ -79,11 +79,13 @@ public static RequestMatcher requestTo(final Matcher<String> matcher) {
 	/**
 	 * Assert the request URI string.
 	 * @param expectedUri the expected URI
+	 * @param uriVars zero or more URI variables to populate the expected URI
 	 * @return the request matcher
 	 */
-	public static RequestMatcher requestTo(final String expectedUri) {
+	public static RequestMatcher requestTo(final String expectedUri, final Object... uriVars) {
 		Assert.notNull(expectedUri, "'uri' must not be null");
-		return request -> assertEquals("Request URI", expectedUri, request.getURI().toString());
+		String uri = UriComponentsBuilder.fromUriString(expectedUri).buildAndExpand(uriVars).encode().toString();
```

Comment:
Yes I see now that a similar change was made not long ago. The ticket also shows a comment about a subsequent regression https://jira.spring.io/browse/SPR-14790 .. :)

The first option isn't really an option because of the inconsistency I pointed out. I shouldn't have called it an option in the first place. Let's go for these two for now:

`requestTo(String)`
`requestToUriTemplate(String, Object...)`

The `requestTouri(String, Function<UriBuilder, URI>)` can always be added later.